### ISSUE TITLE
Grafana Cloud operator output YAML not working

### ISSUE THREAD

**Issue by supergillis** [grafana/agent#3624]
The Grafana Agent Operator configuration instructions generate the following configuration:

```yaml
apiVersion: v1
data: {}
kind: Secret
metadata:
  name: logs-secret
  namespace: default
stringData:
  password: XXX
  username: "123456"
type: Opaque
---
apiVersion: v1
data: {}
kind: Secret
metadata:
  name: metrics-secret
  namespace: default
stringData:
  password: XXX
  username: "123456"
type: Opaque
---
apiVersion: monitoring.grafana.com/v1alpha1
kind: LogsInstance
metadata:
  labels:
    agent: grafana-agent
  name: grafana-agent-logs
  namespace: default
spec:
  clients:
  - basicAuth:
      password:
        key: password
        name: logs-secret
      username:
        key: username
        name: logs-secret
    externalLabels:
      cluster: cloud
    url: https://logs-xxx.grafana.net/loki/api/v1/push
  podLogsNamespaceSelector: {}
  podLogsSelector:
    matchLabels:
      instance: primary
---
apiVersion: monitoring.grafana.com/v1alpha1
kind: MetricsInstance
metadata:
  labels:
    agent: grafana-agent
  name: grafana-agent-metrics
  namespace: default
spec:
  podMonitorNamespaceSelector: {}
  podMonitorSelector:
    matchLabels:
      instance: primary
  remoteWrite:
  - basicAuth:
      password:
        key: password
        name: metrics-secret
      username:
        key: username
        name: metrics-secret
    url: https://prometheus-xxx.grafana.net/api/prom/push
  serviceMonitorNamespaceSelector: {}
  serviceMonitorSelector:
    matchLabels:
      instance: primary
---
```
The generated configuration is deployable but the `MetricsInstance` configuration is not configured correctly. `MetricsInstance` uses `secrets.pathForSecret` and generates a `password_file` field,  but `LogsInstance` uses `secrets.valueForSecret` and generates a `password` field.

Here is the relevant part of the `MetricsInstance` configuration generation: https://github.com/grafana/agent/blob/84a64946c3dcfca5b9ea009322a825ed3a7082be/pkg/operator/config/templates/component/metrics/remote_write.libsonnet#L31-L36

Here is the relevant part of the `LogsInstance` configuration generation: https://github.com/grafana/agent/blob/84a64946c3dcfca5b9ea009322a825ed3a7082be/pkg/operator/config/templates/component/logs/client.libsonnet#L25-L28

It looks like `MetricsInstance` needs to be adjusted to behave the same as the `LogsInstance`.

**Comment by marctc** [comment]
> It looks like `MetricsInstance` needs to be adjusted to behave the same as the `LogsInstance`.

Metrics instance was changed to use only `password_file` (see https://github.com/grafana/agent/issues/1061). I would rephrase saying that `LogInstance` should behave as `MetricsInstances`. @captncraig  do you know why we changed metrics but not logs?

**Comment by supergillis** [comment]
Any update on this? This is blocking us from moving to Grafana Cloud.

**Comment by supergillis** [comment]
You're correct, I tried to reproduce the issue again but it seems everything works as expected now.

Thank you for your time!
